week-04-202402/Headers/Exercise3/Avion.h

```diff
@@ -0,0 +1,153 @@
+#pragma once
+#include "Ala.h"
+#include "Empenaje.h"
+#include "Fuselaje.h"
+#include "TrenAterrizaje.h"
+
+class Avion
+{
+private:
+    float x;
+    float y;
+    float dy;
+    float dx;
+    float velocidad;
+    int color;
+    int alto;
+    int ancho;
+    TrenAterrizaje** trenes;
+    Ala** alas;
+    Fuselaje* fuselaje;
+    Empenaje* empenaje;
+    int numTrenes;
+    int numAlas;
+    int pasoActual;
+    int numPasos;
+    bool haLlegadoFinal;
+
+public:
+    Avion(float x, float y, int color);
+    ~Avion();
+    void Borrar();
+    void Mover();
+    void Dibujar();
+    bool LlegoFinal();
+    bool EstaEstacionado();
+};
+
+Avion::Avion(float x, float y, int color)
+{
+    this->x = x;
+    this->y = y;
+    this->color = color;
+    numPasos = 5;
+    ancho = 9;
+    alto = 2;
+    velocidad = RandFloat(0.1, 1);
+    numTrenes = 2;
+    numAlas = 2;
+    empenaje = new Empenaje(x + 3, y);
+    alas = new Ala * [numAlas];
+    alas[0] = new Ala(x, y + 1);
+    alas[1] = new Ala(x + 6, y + 1);
+    trenes = new TrenAterrizaje * [numTrenes];
+    trenes[0] = new TrenAterrizaje(x + 3, y + 1);
+    trenes[1] = new TrenAterrizaje(x + 5, y + 1);
+    fuselaje = new Fuselaje(x + 4, y + 1);
+    pasoActual = 0;
+    dy = 1;
+    dx = 1;
+    haLlegadoFinal = false;
+}
+
+Avion::~Avion()
+{
+    delete fuselaje;
+    delete empenaje;
+    for (int i = 0; i < numAlas; i++)
```

Comment:
Before going through the for, we could ensure an NPE with the validation
`if (alas != nullptr)`